### DISCUSSION TITLE
Force calculation

### DISCUSSION THREAD

**Discussion by wWL2771140158** [xcompact3d/Incompact3d#158] [Q&A] [answered]
Hello, when I used the Cylinder module to simulate the flow around a cylinder, I placed two cylinders in the calculation domain, but ultimately only the lift and drag coefficients of one cylinder will be output. My calculation domain size is 20Dx10Dx3D, and the centers of the two cylinders are (5D, 5D) and (8D, 5D). How can I output the lift and drag coefficients of the two cylinders?
I would appreciate it if you could help me!
The following is my relevant setting for calculating the lift and drag coefficient!
![Screenshot from 2023-03-07 21-00-23](https://user-images.githubusercontent.com/103632822/223432032-088a5751-1f10-4bb6-9e4e-f8f4899ed1c5.png)

**Comment by slaizet** [comment, marked as answer]
Hi,
You just need to make sure that your -unique- volume control (for the calculation of the forces) contains the two cylinders (keep the number of volumes to 1). It could be from 4D to 9D in X and from 4D to 6D in Y.
You can also implement two volume controls so you can compute the forces for each cylinder (number of volumes to 2, each volume control contains one cylinder). It seems to be what you are doing in your input file.
Drag and lift are saved in forces.dat. With two volume controls, data are saved first for the first volume and then for the second volume (so two lines with the same time step).
I hope this help.
Best,
Sylvain

**Reply by wWL2771140158** [reply]
Yes, I set it according to the method you suggested. I tried both methods, but only one force.dat file will be output in the end, and there is only one cylinder's lift and drag coefficient in this file. I checked force.f90. I think this file has a flaw in the output part of the force coefficient code. I tried to modify it but failed. Can you help me check the output part of this code?

Sorry to bother you again!

**Reply by slaizet** [reply]
There will only be one forces.dat file. As I said, with two volume controls, data are saved first for the first volume and then for the second volume (so two lines with the same time step), see below:
 ```
   2.2499999999999999E-002   18.918193577880274       0.96181467597932968
   2.2499999999999999E-002   15.502927606530070       0.87399811592231169
   2.9999999999999999E-002   10.355157415754819        1.1046568431300694
   2.9999999999999999E-002   8.6789863062699713        1.0445113248746682
   3.7499999999999999E-002   5.9989404506098403        1.2345956820950248
   3.7499999999999999E-002   5.1735211506369492        1.1953972152882699
```
As you can see, you have the same time step (first column) twice, once for the first cylinder and once for the second cylinder.

Best,

Sylvain

**Reply by wWL2771140158** [reply]
Thank you for your patience in answering questions for me. I was too careless. Now I understand what you mean. Thank you again for your help!
